adlfs/tests/test_spec.py

```diff
@@ -1137,6 +1137,22 @@ def test_cat(storage):
     fs.rm("catdir/catfile.txt")
 
 
+def test_url(storage):
```

Comment:
Based on [this](https://stackoverflow.com/questions/50874498/content-disposition-header-ignored-on-download-with-azure-blob-storage), it doesn't appear the tests will work with Azurite.

Reply:
Should I skip the tests altogether then? aka remove them?
Then we would only have the functionality without the tests.
Let me know, Thanks!

Reply:
It would be best if you can create a mock for fs, rather than using Azurite.

Reply:
Seems like I cannot even use just `fs.mkdir("catdir")` aka I removed everything in the test apart from `fs.mkdir` [even calls to `fs.url(...)`]. But even a simple operation such as `fs.mkdir` doesn't work. Like I posted in my first comment, the issue comes when I supply all 3 arguments i.e.

```
fs = AzureBlobFileSystem(
        account_name=ACCOUNT_NAME, account_key=KEY, connection_string=CONN_STR
    )
```
instead of supplying only 2 of them as done in the other tests.

Is this still fixable by mocking?
Also, could you give me some pointers about mocking?
I don't know / haven't tried `mocking` before.

Reply:
Found the issue.  The following fixes it in the unit tests.  You should confirm it works as expected...

```
        async with self.service_client.get_blob_client(container_name, blob) as bc:
            url = f"{bc.url}?{sas_token}"
        return url
```

Reply:
Calling .url on the end was stripping a "/" between the http address and the account name

Reply:
While this doesn't change the behaviour with a live account (which is good), it is still unable to fix the tests for me. I think the reason for tests failing is not because of using `.url`. It's probably because we use all 3 fields i.e. `account_name=ACCOUNT_NAME, account_key=KEY, connection_string=CONN_STR` while instantiating the file system object. Regardless, I have pushed a commit with your suggested change. Let's see if the tests pass on the CI.

Reply:
Seems like the tests passed on the CI. Although they still fail for me locally which is strange. Regardless, this PR should be ready.